test/test_http_adapter.py

```diff
@@ -50,13 +53,29 @@ def test_ssl_verification():
         with pytest.raises(SSLError):
             res = service.send(prepped)
 
+        # Next configure it to validate by using our local certificate. Should raise no exception.
+        res = service.send(prepped, verify=cert)
+        assert res is not None
+
         # Now disable the SSL verification. The request shouldn't raise any issue.
         service.set_disable_ssl_verification(True)
         assert service.disable_ssl_verification is True
         prepped = service.prepare_request('GET', url='/')
         res = service.send(prepped)
         assert res is not None
+
+        # Lastly, try with an external URL.
+        # This test case is mainly here to reproduce the regression
+        # in the `requests` package that was introduced in `2.32.3`.
+        # More details on the issue can be found here: https://github.com/psf/requests/issues/6730
+        service = BaseService(service_url='https://raw.githubusercontent.com', authenticator=NoAuthAuthenticator())
+        assert service.disable_ssl_verification is False
+        prepped = service.prepare_request('GET', url='/IBM/python-sdk-core/main/README.md')
```

Comment:
I'm not too happy with this solution, but I found this to be the most reliable way to reproduce the problem. Actually the only way because I was not able to trigger the exception in any of my local experiments. I totally understand if you have concerns about this, so let's discuss.